pymeasure/adapters/serial.py

```diff
@@ -103,10 +103,9 @@ def _read_bytes(self, count, break_on_termchar, **kwargs):
             return self.connection.read_until(self.read_termination.encode(),
                                               count if count > 0 else None,
                                               **kwargs)
-        elif count == -1:
-            return b"\n".join(self.connection.readlines(**kwargs))
         else:
-            return self.connection.read(count, **kwargs)
+            # At negative count, we read up to one GB, which can be considered the whole buffer.
```

Comment:
I know it makes the code a bit easier, but why make the implementation different from the docstring (and the other Adapters'), with the "negative count" thing?

Reply:
> I exchanged "readlines" from the SerialAdapter (with some drawbacks, see PySerial: Readline) for read_bytes reading 1 GB, as even "readlines" reads until a timeout occurs (maybe several times). We could increase the maximum size, but I figured, that 1 GB is larger than any reasonable chunk size. And if you need more, you can call read_bytes(1e20), for example.

I always found that read lines fishy (reading it on the pyserial page), but it seemed to work as expected.
Having written the tests, it did not work as it should (it added additional term chars) according to the documentation.
As the "read line" documentation said, that it is based on (several times) "read", I transitioned to reading a "real large chunk" as reading everything.
We could increase the byte count to 1e18 or even 1e100, such that nobody will ever read that many bytes in one go.

"negative count" includes - 1

Reply:
Sorry, I meant: the docstring says -1 reads everything,  but in fact -42 now _also_ reads everything,  and that is surprising/inconsistent! And all it seems to gain is one less if clause.

Reply:
Oh, that is what you meant.

I had thought, that it may interpret user input sloppily. I'll change it for strict comparison.

Reply:
thanks!